language/vm/vm-runtime/src/chain_state.rs

```diff
@@ -120,3 +150,70 @@ impl<'txn> ChainState for TransactionExecutionContext<'txn> {
         self.event_data.push(event)
     }
 }
+
+pub struct GasFreeContext<'txn>(TransactionExecutionContext<'txn>);
```

Comment:
I wonder if something like `SystemExecutionContext` or similar is a better name for this. 
Not a blocker just trying to come up with better names